test/e2e/helmreleaseproxy_helpers.go

```diff
@@ -88,3 +100,33 @@ func getHelmReleaseProxy(
 
 	return &releaseList.Items[0], nil
 }
+
+func getHelmChartProxy(
+	ctx context.Context,
+	getLister framework.GetLister,
+	clusterName string,
+	clusterNamespace string,
+	helmReleaseName string,
+) (*helmaddonsv1.HelmChartProxy, error) {
+	// Get the HelmChartProxy using label selectors since we don't know the name of the HelmChartProxy.
+	chartList := &helmaddonsv1.HelmChartProxyList{}
+	labels := map[string]string{
+		addons.ClusterNamespaceLabel:    clusterNamespace,
+		clusterv1.ClusterNameLabel:      clusterName,
+		addons.HelmReleaseNameHashLabel: fmt.Sprintf("%x", sha256.Sum224([]byte(helmReleaseName))),
```

Comment:
nit: Add some context.

```suggestion
        // Reflects the hash implementation used by the helm addon strategy.
		addons.HelmReleaseNameHashLabel: fmt.Sprintf("%x", sha256.Sum224([]byte(helmReleaseName))),
```